nim: 0.18.0 -> 0.19.0

royneary opened the pull request:
- remove sqlite as dependency (I don't see why it's needed)
- run checkPhase after installPhase (at least one test assumes the
standard library in ../lib relative to the nim binary)
- the broken tests pass now or don't exist anymore
- two of the tests requiring network access pass now without network access
- the tests in manyloc do not download dependencies and are passing now

###### Motivation for this change
Update nim to 0.19.0.

###### Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

royneary committed:
nim: 0.18.0 -> 0.19.0

- remove sqlite as dependency (I don't see why it's needed)
- run checkPhase after installPhase (at least one test assumes the
standard library in ../lib relative to the nim binary)
- the broken tests pass now or don't exist anymore
- two of the tests requiring network access pass now without network access
- the tests in manyloc do not download dependencies and are passing now

disassembler commented:
@GrahamcOfBorg build nim

GrahamcOfBorg commented:
<!--REQUEST_ID=3144db11-f799-4dae-ae35-4ec52ad12bc7-->
Failure on aarch64-linux [(full log)](https://logs.nix.ci/?key=nixos/nixpkgs.49250&attempt_id=f5a53b8e-3f48-4d59-8732-fe7d9a333d07)

Attempted: nim

<details><summary>Partial log (click to expand)</summary><p>

```
PASS: cyclecollector C -d:release                                  (3.40048289 secs)
PASS: cyclecollector C -d:release -d:useRealtimeGC                 (3.43674994 secs)
PASS: cyclecollector C --gc:markAndSweep                           (2.61317205 secs)
PASS: cyclecollector C -d:release --gc:markAndSweep                (2.97520089 secs)
PASS: cyclecollector C --gc:boehm                                  (1.25764084 secs)
PASS: cyclecollector C -d:release --gc:boehm                       (0.91316295 secs)
/build/nim-0.19.0/tests/testament/tester html
tests failed
builder for '/nix/store/ci71mgmf37hb10k3pjyd5dblak95wy2i-nim-0.19.0.drv' failed with exit code 1
error: build of '/nix/store/ci71mgmf37hb10k3pjyd5dblak95wy2i-nim-0.19.0.drv' failed
```
</p></details>

GrahamcOfBorg commented:
<!--REQUEST_ID=3144db11-f799-4dae-ae35-4ec52ad12bc7-->
Timed out, unknown build status on x86_64-darwin [(full log)](https://logs.nix.ci/?key=nixos/nixpkgs.49250&attempt_id=9eb575ac-c597-4fe5-91ab-4646938e7aa1)

Attempted: nim

<details><summary>Partial log (click to expand)</summary><p>

```
  clang++ '-DV8_GYP_BUILD' '-D_DARWIN_USE_64_BIT_INODE=1' '-DV8_TARGET_ARCH_X64' '-DENABLE_DISASSEMBLER' '-DV8_PROMISE_INTERNAL_FIELD_COUNT' '-Dv8_promise_internal_field_count' '-DV8_INTL_SUPPORT' '-DICU_UTIL_DATA_IMPL=ICU_UTIL_DATA_STATIC' '-DUCONFIG_NO_SERVICE=1' '-DUCONFIG_NO_REGULAR_EXPRESSIONS=1' '-DU_ENABLE_DYLOAD=0' '-DU_STATIC_IMPLEMENTATION=1' '-DU_HAVE_STD_STRING=0' '-DUCONFIG_NO_BREAK_ITERATION=0' -I../deps/v8 -I../. -I/private/tmp/nix-build-nodejs-slim-8.12.0.drv-0/node-v8.12.0/out/Release/obj/gen -I/nix/store/qscisgacfgykr63pgmjp1s65rndma3wy-zlib-1.2.11-dev/include -I/nix/store/59c6bz08jxcyxyi962fhf56jzph3f333-libuv-1.23.1/include -I/nix/store/j3cqsdyfdzv21kkgzwv19f7cxzz00znz-openssl-1.0.2p-dev/include -I../deps/v8/include -I../deps/icu-small/source/i18n -I../deps/icu-small/source/common  -O3 -gdwarf-2 -fstrict-aliasing -mmacosx-version-min=10.7 -arch x86_64 -Wall -Wendif-labels -W -Wno-unused-parameter -std=gnu++0x -stdlib=libc++ -fno-rtti -fno-exceptions -fno-threadsafe-statics -fno-strict-aliasing -MMD -MF /private/tmp/nix-build-nodejs-slim-8.12.0.drv-0/node-v8.12.0/out/Release/.deps//private/tmp/nix-build-nodejs-slim-8.12.0.drv-0/node-v8.12.0/out/Release/obj.target/v8_base/deps/v8/src/compiler/js-create-lowering.o.d.raw   -c -o /private/tmp/nix-build-nodejs-slim-8.12.0.drv-0/node-v8.12.0/out/Release/obj.target/v8_base/deps/v8/src/compiler/js-create-lowering.o ../deps/v8/src/compiler/js-create-lowering.cc
  clang++ '-DV8_GYP_BUILD' '-D_DARWIN_USE_64_BIT_INODE=1' '-DV8_TARGET_ARCH_X64' '-DENABLE_DISASSEMBLER' '-DV8_PROMISE_INTERNAL_FIELD_COUNT' '-Dv8_promise_internal_field_count' '-DV8_INTL_SUPPORT' '-DICU_UTIL_DATA_IMPL=ICU_UTIL_DATA_STATIC' '-DUCONFIG_NO_SERVICE=1' '-DUCONFIG_NO_REGULAR_EXPRESSIONS=1' '-DU_ENABLE_DYLOAD=0' '-DU_STATIC_IMPLEMENTATION=1' '-DU_HAVE_STD_STRING=0' '-DUCONFIG_NO_BREAK_ITERATION=0' -I../deps/v8 -I../. -I/private/tmp/nix-build-nodejs-slim-8.12.0.drv-0/node-v8.12.0/out/Release/obj/gen -I/nix/store/qscisgacfgykr63pgmjp1s65rndma3wy-zlib-1.2.11-dev/include -I/nix/store/59c6bz08jxcyxyi962fhf56jzph3f333-libuv-1.23.1/include -I/nix/store/j3cqsdyfdzv21kkgzwv19f7cxzz00znz-openssl-1.0.2p-dev/include -I../deps/v8/include -I../deps/icu-small/source/i18n -I../deps/icu-small/source/common  -O3 -gdwarf-2 -fstrict-aliasing -mmacosx-version-min=10.7 -arch x86_64 -Wall -Wendif-labels -W -Wno-unused-parameter -std=gnu++0x -stdlib=libc++ -fno-rtti -fno-exceptions -fno-threadsafe-statics -fno-strict-aliasing -MMD -MF /private/tmp/nix-build-nodejs-slim-8.12.0.drv-0/node-v8.12.0/out/Release/.deps//private/tmp/nix-build-nodejs-slim-8.12.0.drv-0/node-v8.12.0/out/Release/obj.target/v8_base/deps/v8/src/compiler/js-generic-lowering.o.d.raw   -c -o /private/tmp/nix-build-nodejs-slim-8.12.0.drv-0/node-v8.12.0/out/Release/obj.target/v8_base/deps/v8/src/compiler/js-generic-lowering.o ../deps/v8/src/compiler/js-generic-lowering.cc
  clang++ '-DV8_GYP_BUILD' '-D_DARWIN_USE_64_BIT_INODE=1' '-DV8_TARGET_ARCH_X64' '-DENABLE_DISASSEMBLER' '-DV8_PROMISE_INTERNAL_FIELD_COUNT' '-Dv8_promise_internal_field_count' '-DV8_INTL_SUPPORT' '-DICU_UTIL_DATA_IMPL=ICU_UTIL_DATA_STATIC' '-DUCONFIG_NO_SERVICE=1' '-DUCONFIG_NO_REGULAR_EXPRESSIONS=1' '-DU_ENABLE_DYLOAD=0' '-DU_STATIC_IMPLEMENTATION=1' '-DU_HAVE_STD_STRING=0' '-DUCONFIG_NO_BREAK_ITERATION=0' -I../deps/v8 -I../. -I/private/tmp/nix-build-nodejs-slim-8.12.0.drv-0/node-v8.12.0/out/Release/obj/gen -I/nix/store/qscisgacfgykr63pgmjp1s65rndma3wy-zlib-1.2.11-dev/include -I/nix/store/59c6bz08jxcyxyi962fhf56jzph3f333-libuv-1.23.1/include -I/nix/store/j3cqsdyfdzv21kkgzwv19f7cxzz00znz-openssl-1.0.2p-dev/include -I../deps/v8/include -I../deps/icu-small/source/i18n -I../deps/icu-small/source/common  -O3 -gdwarf-2 -fstrict-aliasing -mmacosx-version-min=10.7 -arch x86_64 -Wall -Wendif-labels -W -Wno-unused-parameter -std=gnu++0x -stdlib=libc++ -fno-rtti -fno-exceptions -fno-threadsafe-statics -fno-strict-aliasing -MMD -MF /private/tmp/nix-build-nodejs-slim-8.12.0.drv-0/node-v8.12.0/out/Release/.deps//private/tmp/nix-build-nodejs-slim-8.12.0.drv-0/node-v8.12.0/out/Release/obj.target/v8_base/deps/v8/src/compiler/js-graph.o.d.raw   -c -o /private/tmp/nix-build-nodejs-slim-8.12.0.drv-0/node-v8.12.0/out/Release/obj.target/v8_base/deps/v8/src/compiler/js-graph.o ../deps/v8/src/compiler/js-graph.cc
  clang++ '-DV8_GYP_BUILD' '-D_DARWIN_USE_64_BIT_INODE=1' '-DV8_TARGET_ARCH_X64' '-DENABLE_DISASSEMBLER' '-DV8_PROMISE_INTERNAL_FIELD_COUNT' '-Dv8_promise_internal_field_count' '-DV8_INTL_SUPPORT' '-DICU_UTIL_DATA_IMPL=ICU_UTIL_DATA_STATIC' '-DUCONFIG_NO_SERVICE=1' '-DUCONFIG_NO_REGULAR_EXPRESSIONS=1' '-DU_ENABLE_DYLOAD=0' '-DU_STATIC_IMPLEMENTATION=1' '-DU_HAVE_STD_STRING=0' '-DUCONFIG_NO_BREAK_ITERATION=0' -I../deps/v8 -I../. -I/private/tmp/nix-build-nodejs-slim-8.12.0.drv-0/node-v8.12.0/out/Release/obj/gen -I/nix/store/qscisgacfgykr63pgmjp1s65rndma3wy-zlib-1.2.11-dev/include -I/nix/store/59c6bz08jxcyxyi962fhf56jzph3f333-libuv-1.23.1/include -I/nix/store/j3cqsdyfdzv21kkgzwv19f7cxzz00znz-openssl-1.0.2p-dev/include -I../deps/v8/include -I../deps/icu-small/source/i18n -I../deps/icu-small/source/common  -O3 -gdwarf-2 -fstrict-aliasing -mmacosx-version-min=10.7 -arch x86_64 -Wall -Wendif-labels -W -Wno-unused-parameter -std=gnu++0x -stdlib=libc++ -fno-rtti -fno-exceptions -fno-threadsafe-statics -fno-strict-aliasing -MMD -MF /private/tmp/nix-build-nodejs-slim-8.12.0.drv-0/node-v8.12.0/out/Release/.deps//private/tmp/nix-build-nodejs-slim-8.12.0.drv-0/node-v8.12.0/out/Release/obj.target/v8_base/deps/v8/src/compiler/js-inlining.o.d.raw   -c -o /private/tmp/nix-build-nodejs-slim-8.12.0.drv-0/node-v8.12.0/out/Release/obj.target/v8_base/deps/v8/src/compiler/js-inlining.o ../deps/v8/src/compiler/js-inlining.cc
  clang++ '-DV8_GYP_BUILD' '-D_DARWIN_USE_64_BIT_INODE=1' '-DV8_TARGET_ARCH_X64' '-DENABLE_DISASSEMBLER' '-DV8_PROMISE_INTERNAL_FIELD_COUNT' '-Dv8_promise_internal_field_count' '-DV8_INTL_SUPPORT' '-DICU_UTIL_DATA_IMPL=ICU_UTIL_DATA_STATIC' '-DUCONFIG_NO_SERVICE=1' '-DUCONFIG_NO_REGULAR_EXPRESSIONS=1' '-DU_ENABLE_DYLOAD=0' '-DU_STATIC_IMPLEMENTATION=1' '-DU_HAVE_STD_STRING=0' '-DUCONFIG_NO_BREAK_ITERATION=0' -I../deps/v8 -I../. -I/private/tmp/nix-build-nodejs-slim-8.12.0.drv-0/node-v8.12.0/out/Release/obj/gen -I/nix/store/qscisgacfgykr63pgmjp1s65rndma3wy-zlib-1.2.11-dev/include -I/nix/store/59c6bz08jxcyxyi962fhf56jzph3f333-libuv-1.23.1/include -I/nix/store/j3cqsdyfdzv21kkgzwv19f7cxzz00znz-openssl-1.0.2p-dev/include -I../deps/v8/include -I../deps/icu-small/source/i18n -I../deps/icu-small/source/common  -O3 -gdwarf-2 -fstrict-aliasing -mmacosx-version-min=10.7 -arch x86_64 -Wall -Wendif-labels -W -Wno-unused-parameter -std=gnu++0x -stdlib=libc++ -fno-rtti -fno-exceptions -fno-threadsafe-statics -fno-strict-aliasing -MMD -MF /private/tmp/nix-build-nodejs-slim-8.12.0.drv-0/node-v8.12.0/out/Release/.deps//private/tmp/nix-build-nodejs-slim-8.12.0.drv-0/node-v8.12.0/out/Release/obj.target/v8_base/deps/v8/src/compiler/js-inlining-heuristic.o.d.raw   -c -o /private/tmp/nix-build-nodejs-slim-8.12.0.drv-0/node-v8.12.0/out/Release/obj.target/v8_base/deps/v8/src/compiler/js-inlining-heuristic.o ../deps/v8/src/compiler/js-inlining-heuristic.cc
  clang++ '-DV8_GYP_BUILD' '-D_DARWIN_USE_64_BIT_INODE=1' '-DV8_TARGET_ARCH_X64' '-DENABLE_DISASSEMBLER' '-DV8_PROMISE_INTERNAL_FIELD_COUNT' '-Dv8_promise_internal_field_count' '-DV8_INTL_SUPPORT' '-DICU_UTIL_DATA_IMPL=ICU_UTIL_DATA_STATIC' '-DUCONFIG_NO_SERVICE=1' '-DUCONFIG_NO_REGULAR_EXPRESSIONS=1' '-DU_ENABLE_DYLOAD=0' '-DU_STATIC_IMPLEMENTATION=1' '-DU_HAVE_STD_STRING=0' '-DUCONFIG_NO_BREAK_ITERATION=0' -I../deps/v8 -I../. -I/private/tmp/nix-build-nodejs-slim-8.12.0.drv-0/node-v8.12.0/out/Release/obj/gen -I/nix/store/qscisgacfgykr63pgmjp1s65rndma3wy-zlib-1.2.11-dev/include -I/nix/store/59c6bz08jxcyxyi962fhf56jzph3f333-libuv-1.23.1/include -I/nix/store/j3cqsdyfdzv21kkgzwv19f7cxzz00znz-openssl-1.0.2p-dev/include -I../deps/v8/include -I../deps/icu-small/source/i18n -I../deps/icu-small/source/common  -O3 -gdwarf-2 -fstrict-aliasing -mmacosx-version-min=10.7 -arch x86_64 -Wall -Wendif-labels -W -Wno-unused-parameter -std=gnu++0x -stdlib=libc++ -fno-rtti -fno-exceptions -fno-threadsafe-statics -fno-strict-aliasing -MMD -MF /private/tmp/nix-build-nodejs-slim-8.12.0.drv-0/node-v8.12.0/out/Release/.deps//private/tmp/nix-build-nodejs-slim-8.12.0.drv-0/node-v8.12.0/out/Release/obj.target/v8_base/deps/v8/src/compiler/js-intrinsic-lowering.o.d.raw   -c -o /private/tmp/nix-build-nodejs-slim-8.12.0.drv-0/node-v8.12.0/out/Release/obj.target/v8_base/deps/v8/src/compiler/js-intrinsic-lowering.o ../deps/v8/src/compiler/js-intrinsic-lowering.cc
  clang++ '-DV8_GYP_BUILD' '-D_DARWIN_USE_64_BIT_INODE=1' '-DV8_TARGET_ARCH_X64' '-DENABLE_DISASSEMBLER' '-DV8_PROMISE_INTERNAL_FIELD_COUNT' '-Dv8_promise_internal_field_count' '-DV8_INTL_SUPPORT' '-DICU_UTIL_DATA_IMPL=ICU_UTIL_DATA_STATIC' '-DUCONFIG_NO_SERVICE=1' '-DUCONFIG_NO_REGULAR_EXPRESSIONS=1' '-DU_ENABLE_DYLOAD=0' '-DU_STATIC_IMPLEMENTATION=1' '-DU_HAVE_STD_STRING=0' '-DUCONFIG_NO_BREAK_ITERATION=0' -I../deps/v8 -I../. -I/private/tmp/nix-build-nodejs-slim-8.12.0.drv-0/node-v8.12.0/out/Release/obj/gen -I/nix/store/qscisgacfgykr63pgmjp1s65rndma3wy-zlib-1.2.11-dev/include -I/nix/store/59c6bz08jxcyxyi962fhf56jzph3f333-libuv-1.23.1/include -I/nix/store/j3cqsdyfdzv21kkgzwv19f7cxzz00znz-openssl-1.0.2p-dev/include -I../deps/v8/include -I../deps/icu-small/source/i18n -I../deps/icu-small/source/common  -O3 -gdwarf-2 -fstrict-aliasing -mmacosx-version-min=10.7 -arch x86_64 -Wall -Wendif-labels -W -Wno-unused-parameter -std=gnu++0x -stdlib=libc++ -fno-rtti -fno-exceptions -fno-threadsafe-statics -fno-strict-aliasing -MMD -MF /private/tmp/nix-build-nodejs-slim-8.12.0.drv-0/node-v8.12.0/out/Release/.deps//private/tmp/nix-build-nodejs-slim-8.12.0.drv-0/node-v8.12.0/out/Release/obj.target/v8_base/deps/v8/src/compiler/js-native-context-specialization.o.d.raw   -c -o /private/tmp/nix-build-nodejs-slim-8.12.0.drv-0/node-v8.12.0/out/Release/obj.target/v8_base/deps/v8/src/compiler/js-native-context-specialization.o ../deps/v8/src/compiler/js-native-context-specialization.cc
building of '/nix/store/gdcvliy19dcvcsrckg687jk1b82mhqq0-nodejs-slim-8.12.0.drv' timed out after 1800 seconds
cannot build derivation '/nix/store/msxydnlij99p5m7k08ja42g846dbwjl0-nim-0.19.0.drv': 1 dependencies couldn't be built
error: build of '/nix/store/msxydnlij99p5m7k08ja42g846dbwjl0-nim-0.19.0.drv' failed
```
</p></details>

GrahamcOfBorg commented:
<!--REQUEST_ID=3144db11-f799-4dae-ae35-4ec52ad12bc7-->
Timed out, unknown build status on x86_64-linux [(full log)](https://logs.nix.ci/?key=nixos/nixpkgs.49250&attempt_id=0e405725-e344-4882-b4d1-98ae2ded553d)

Attempted: nim

<details><summary>Partial log (click to expand)</summary><p>

```
PASS: cycleleak C -d:release --gc:markAndSweep                     (3.16289592 secs)
PASS: cycleleak C --gc:boehm                                       (1.32490492 secs)
PASS: cycleleak C -d:release --gc:boehm                            (1.19588614 secs)
PASS: closureleak C                                                (3.68987894 secs)
PASS: closureleak C -d:release                                     (3.80776811 secs)
PASS: closureleak C -d:release -d:useRealtimeGC                    (3.86992311 secs)
PASS: closureleak C --gc:markAndSweep                              (3.14992309 secs)
PASS: closureleak C -d:release --gc:markAndSweep                   (3.86592388 secs)
building of '/nix/store/n834y56z1brj3ml7npm1zpyfm71ijxcy-nim-0.19.0.drv' timed out after 1800 seconds
error: build of '/nix/store/n834y56z1brj3ml7npm1zpyfm71ijxcy-nim-0.19.0.drv' failed
```
</p></details>

royneary committed:
nim: don't hardcode gcc

royneary committed:
nim: disable test supposedly broken on Aarch64

royneary commented:
I tried to fix the failure on Aarch64.
What can we do about the timeouts?
- On x86_64 the tests were interrupted after 1800 seconds, I don't think they were hanging.
- On darwin building nodejs took too long. I could try to disable nodejs for darwin, it's only needed for the tests that are not run anyway.

Mic92 commented:
If ofborg timeouts this is ok. Our hydra has longer timeouts.

Mic92 commented:
@GrahamcOfBorg build nim

GrahamcOfBorg commented:
<!--REQUEST_ID=809ffa0c-8248-4cb0-9350-66bc303ebbc5-->
Success on aarch64-linux [(full log)](https://logs.nix.ci/?key=nixos/nixpkgs.49250&attempt_id=2aad46c5-99cd-4962-b7c1-78829f562a1c)

Attempted: nim

<details><summary>Partial log (click to expand)</summary><p>

```
PASS: stackrefleak C -d:release --gc:boehm                         (1.88382483 secs)
PASS: cyclecollector C                                             (3.06635785 secs)
PASS: cyclecollector C -d:release                                  (3.41082907 secs)
PASS: cyclecollector C -d:release -d:useRealtimeGC                 (3.43448997 secs)
PASS: cyclecollector C --gc:markAndSweep                           (2.60068607 secs)
PASS: cyclecollector C -d:release --gc:markAndSweep                (2.96925998 secs)
PASS: cyclecollector C --gc:boehm                                  (1.24932003 secs)
PASS: cyclecollector C -d:release --gc:boehm                       (0.91260719 secs)
/build/nim-0.19.0/tests/testament/tester html
/nix/store/7nxv376k3ry7khwmrd5axy7mbn8nnnlg-nim-0.19.0
```
</p></details>

GrahamcOfBorg commented:
<!--REQUEST_ID=809ffa0c-8248-4cb0-9350-66bc303ebbc5-->
Timed out, unknown build status on x86_64-linux [(full log)](https://logs.nix.ci/?key=nixos/nixpkgs.49250&attempt_id=828d9b0a-e14a-4081-977b-b7f9839f362d)

Attempted: nim

<details><summary>Partial log (click to expand)</summary><p>

```
PASS: unicode.nim C                                                (3.06197500 secs)
PASS: unittest.nim C                                               (2.31589603 secs)
PASS: uri.nim C                                                    (2.17861605 secs)
PASS: volatile.nim C                                               (1.59678316 secs)
PASS: xmldom.nim C                                                 (1.88742185 secs)
PASS: xmldomparser.nim C                                           (2.13288188 secs)
PASS: xmlparser.nim C                                              (2.51197982 secs)
PASS: xmltree.nim C                                                (2.50686479 secs)
building of '/nix/store/838ldag9fs6p9lvsajhqm3vw8v1fz1ly-nim-0.19.0.drv' timed out after 1800 seconds
error: build of '/nix/store/838ldag9fs6p9lvsajhqm3vw8v1fz1ly-nim-0.19.0.drv' failed
```
</p></details>

c0bw3b commented on pkgs/development/compilers/nim/default.nix:
```diff
@@ -30,10 +29,16 @@ stdenv.mkDerivation rec {
 
   buildInputs = [
     makeWrapper nodejs-slim-8_x tzdata coreutils
```
`makeWrapper tzdata coreutils` should be in `nativeBuildInputs` (they're needed only in installPhase and checkPhase)

c0bw3b commented:
Build + tests successful on nixos-18.09 otherwise
[Full log here](https://gist.github.com/c0bw3b/798bb7b9ac79043bd957e79ed20457cd)

royneary committed:
nim: move build time dependencies into nativeBuildInputs

royneary committed:
nim: update nodejs-slim

royneary commented:
thanks for reviewing. I moved the packages you mentioned to `nativeBuildInputs` and `nodejs-slim`, too because it's only needed for the `checkPhase`. 

Also I updated `nodejs-slim-8_x` to `nodejs-slim-10_x`.

gebner commented:
@royneary You need to add the `HOME=$TMPDIR` fix to the other nim packages as well.

@GrahamcOfBorg build nrpl

(`mosdepth` doesn't seem to compile with the new version.)

GrahamcOfBorg commented:
<!--REQUEST_ID=4bd1e65e-60c9-47ab-b14d-4595039ee768-->
No attempt on aarch64-linux [(full log)](https://logs.nix.ci/?key=nixos/nixpkgs.49250&attempt_id=30366961-4f31-4717-b7bd-2a60212b02bd)

The following builds were skipped because they don't evaluate on aarch64-linux: nrpl

<details><summary>Partial log (click to expand)</summary><p>

```

a) For `nixos-rebuild` you can set
  { nixpkgs.config.allowUnsupportedSystem = true; }
in configuration.nix to override this.

b) For `nix-env`, `nix-build`, `nix-shell` or any other Nix command you can add
  { allowUnsupportedSystem = true; }
to ~/.config/nixpkgs/config.nix.


```
</p></details>

GrahamcOfBorg commented:
<!--REQUEST_ID=4bd1e65e-60c9-47ab-b14d-4595039ee768-->
Timed out, unknown build status on x86_64-linux [(full log)](https://logs.nix.ci/?key=nixos/nixpkgs.49250&attempt_id=e5c6f48d-4f1c-443e-b14d-6aa5eb53afdb)

Attempted: nrpl

<details><summary>Partial log (click to expand)</summary><p>

```
PASS: unittest.nim C                                               (2.39380908 secs)
PASS: uri.nim C                                                    (2.15112114 secs)
PASS: volatile.nim C                                               (1.61674714 secs)
PASS: xmldom.nim C                                                 (1.96292019 secs)
PASS: xmldomparser.nim C                                           (2.10824990 secs)
PASS: xmlparser.nim C                                              (2.68674994 secs)
PASS: xmltree.nim C                                                (2.47554517 secs)
building of '/nix/store/p48w1n52q5nl4xbs8frwa4ppxmxsnn20-nim-0.19.0.drv' timed out after 1800 seconds
cannot build derivation '/nix/store/swd0ggw35z30miqx2qkhk2vbwri7gnbs-nrpl-20150522.drv': 1 dependencies couldn't be built
error: build of '/nix/store/swd0ggw35z30miqx2qkhk2vbwri7gnbs-nrpl-20150522.drv' failed
```
</p></details>

royneary committed:
mosdeph: fix compilation with nim 0.19

royneary committed:
nrpl: fix compilation with nim 0.19

royneary commented:
@gebner fixed.

gebner commented:
`nrpl` builds for me now.
